File: Leetcode/0228 Summary Ranges.py
class Solution:
    def summaryRanges(self, nums):
        n = len(nums)
        if n == 0:
            return []
        ranges = []
        a, b = nums[0], nums[0]
        for num in nums:
            if num <= (b + 1):
                b += int(num == (b + 1))
            else:
                ranges.append(f'{a}' if a == b else f'{a}->{b}')
                a, b = num, num
        ranges.append(f'{a}' if a == b else f'{a}->{b}')
        return ranges

File: Leetcode/0228 Summary Ranges.ts
function summaryRanges(nums: number[]): string[] {
    if (nums.length === 0) {
        return [];
    }
    let start = nums[0], end = nums[0];
    const ranges: string[] = [];
    for (const num of nums) {
        if (num <= (end + 1)) {
            end += num === end + 1? 1 : 0;
        } else {
            ranges.push(start === end ? `${start}` : `${start}->${end}`);
            start = end = num;
        }
    }
    ranges.push(start === end ? `${start}` : `${start}->${end}`);
    return ranges;
}
